src/org/opendatakit/briefcase/ui/automation/AutomationPanelForm.java

```diff
@@ -0,0 +1,138 @@
+package org.opendatakit.briefcase.ui.automation;
+
+import static org.opendatakit.briefcase.ui.reused.FileChooser.directory;
+
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import org.opendatakit.briefcase.util.StringUtils;
+
+@SuppressWarnings("checkstyle:MethodName")
+public class AutomationPanelForm {
+  public JPanel container;
+  private JLabel scriptDirLabel;
+  private JTextField scriptDirField;
+  private JButton scriptDirChooseButton;
+  private JButton generateScriptButton;
+
+  public AutomationPanelForm() {
+    $$$setupUI$$$();
+    scriptDirChooseButton.addActionListener(__ ->
+        directory(container, fileFrom(scriptDirField))
+            .choose()
+            .ifPresent(file -> setScriptDir(Paths.get(file.toURI()))));
+    generateScriptButton.addActionListener(__ -> {
+      if (isOnWindows()) {
+        generateScript("automation.bat", "echo %JAVA_HOME%");
+      } else {
+        generateScript("automation.sh", "echo $JAVA_HOME");
+      }
+    });
+  }
+
+  private void setScriptDir(Path path) {
+    scriptDirField.setText(path.toString());
+  }
+
+  private static Optional<File> fileFrom(JTextField textField) {
+    return Optional.ofNullable(textField.getText())
+        .filter(StringUtils::nullOrEmpty)
+        .map(path -> Paths.get(path).toFile());
+  }
+
+  private void generateScript(String fileName, String contents) {
+    File script = new File(scriptDirField.getText(), fileName);
```

Comment:
We should always use the Java NIO2 API. Change this with `Files.write(Path path, byte[] bytes, OpenOption... options)`